src/ListView.vala

```diff
@@ -280,18 +289,24 @@ public class Tasks.ListView : Gtk.Grid {
         var row_b = (Tasks.TaskRow) row2;
 
         if (row_a.completed == row_b.completed) {
-            var apple_sortorder_a = Util.get_apple_sortorder_property_value (row_a.task);
-            if (apple_sortorder_a == null) {
-                apple_sortorder_a = Util.get_apple_sortorder_default_value (row_a.task).as_int ().to_string ();
-            }
-
-            var apple_sortorder_b = Util.get_apple_sortorder_property_value (row_b.task);
-            if (apple_sortorder_b == null) {
-                apple_sortorder_b = Util.get_apple_sortorder_default_value (row_b.task).as_int ().to_string ();
-            }
+            if (is_gtasks) {
+                var gtask_position_a = Util.get_gtasks_position_property_value (row_a.task);
+                var gtask_position_b = Util.get_gtasks_position_property_value (row_b.task);
+
+                return gtask_position_a.collate (gtask_position_b);
```

Comment:
How should we deal with `gtask_position_a` or `gtask_position_b` beeing `null` here?

Reply:
I don't think they can be null, if a task is created without setting a position, it's settled to be on the top of the tasklist, however, looks like evolution isn't updating the position for already known tasks

Reply:
So i found the issue, the GTasks API doesn't count moving the task as a update, so EDS don't update the task if the only change is on position, we probably need to patch it on the system EDS rather the flatpak one because the backends are on the system rather than the sandbox, otherwise we need to get the position using libgdata directly.

Reply:
If you not already did, can you open an issue upstream so it is fixed in a future version of EDS?

In terms of elementary, its unlikely we are going to backport EDS - therefore we need to go for the workaround, at least on Odin.

Reply:
also, I'd prefer we make sure we are not running into potential `null` errors and set `gtask_position_{a,b}` to an arbitrary value in case it is really returned as null - just to be sure.

Reply:
> If you not already did, can you open an issue upstream so it is fixed in a future version of EDS?

Upstream issue [opened here](https://gitlab.gnome.org/GNOME/evolution-data-server/-/issues/322).
> In terms of elementary, its unlikely we are going to backport EDS - therefore we need to go for the workaround, at least on Odin.

We probably want to do that with #200, and use them for automatic refresh instead of relying on the eds one, or at least trigger the position update on the same interval of the eds one.

> also, I'd prefer we make sure we are not running into potential `null` errors and set `gtask_position_{a,b}` to an arbitrary value in case it is really returned as null - just to be sure.

I updated the function to ever return a value, if we can't get the property, it's will return the default value of a task created without setting a position.